mne/preprocessing/ecg.py

```diff
@@ -178,28 +184,65 @@ def find_ecg_events(raw, event_id=999, ch_name=None, tstart=0.0,
     create_ecg_epochs
     compute_proj_ecg
     """
+    if reject_by_annotation is None:
+        if len(raw.annotations) > 0:
+            warn('reject_by_annotation in find_ecg_events defaults to False '
+                 'in 0.18 but will change to True in 0.19, set it explicitly '
+                 'to avoid this warning', DeprecationWarning)
+        reject_by_annotation = False
+    skip_by_annotation = ('edge', 'bad') if reject_by_annotation else ()
+    del reject_by_annotation
     idx_ecg = _get_ecg_channel_index(ch_name, raw)
     if idx_ecg is not None:
         logger.info('Using channel %s to identify heart beats.'
                     % raw.ch_names[idx_ecg])
-        ecg, times = raw[idx_ecg, :]
+        ecg = raw.get_data(picks=idx_ecg)
     else:
-        ecg, times = _make_ecg(raw, None, None, verbose=verbose)
+        ecg = _make_ecg(raw, None, None)[0]
+    assert ecg.ndim == 2 and ecg.shape[0] == 1
+    ecg = ecg[0]
+    # Deal with filtering the same way we do in raw, i.e. filter each good
+    # segment
+    onsets, ends = _annotations_starts_stops(
+        raw, skip_by_annotation, 'reject_by_annotation', invert=True)
+    ecgs = list()
+    max_idx = (ends - onsets).argmax()
+    for si, (start, stop) in enumerate(zip(onsets, ends)):
+        use_verbose = verbose if si == max_idx else 'error'
```

Comment:
I am lost here too. Why this verbose?

Reply:
This way if you have 100 segments, you only get 1 warning about a filter length being longer than your longest segment, instead of possibly 100 warnings.